pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -63,15 +63,10 @@ func ConvertService(svc coreV1.Service, domainSuffix string, clusterID string) *
 	meshExternal := false
 
 	if svc.Spec.Type == coreV1.ServiceTypeExternalName && svc.Spec.ExternalName != "" {
-		external = svc.Spec.ExternalName
 		resolution = model.DNSLB
 		meshExternal = true
 	}
 
-	if addr == constants.UnspecifiedIP && external == "" { // headless services should not be load balanced
-		resolution = model.Passthrough
-	}
```

Comment:
This may work for redis but will not work for other things like elasticsearch or etcd or zookeeper etc., where the idea is that the user is explicitly connecting to a pod in the headless service. Also, it won't work between the pods of the headless service as they try to talk to other specific pods inn the headless service for database clustering.

Reply:
I donot think so, i have also tested by setting the portname to tcp. And it works.